### ISSUE TITLE
_serialize method has polymorph half-asynchronous / half-synchronous behaviour.

### ISSUE THREAD

**Issue by lu4** [localForage/localForage#317]
There is a problem deeply buried within `_serialize` method. When trying to work with `[object Blob]` objects.

The `_serialize` method will start behaving in async fashion. In all other cases `_serialize` method will behave in sync fashion. 

The problem is that localStorage driver is implemented totally in sync fashion (via `return ...` rather than `resolve`) it will cause each operation to resolve earlier then it's actual completion.

In other words LocalStorage doesn't work with `[object Blob]` type of objects. I.e it is not possible to be sure whether the operation is finished or not, was it successful or not...
